src/components/CardGrid/CardGrid.js

```diff
@@ -1,10 +1,20 @@
 import StyledFlex from '../../../styles/StyledFlex';
+import useStore from '../../lib/hooks/useStore';
 import Card from '../Card/Card';
+import CarFilter from '../CarFilter/CarFilter';
 
 export default function CardGrid() {
+	const filterState = useStore(state => state.filterState);
+	const myVehicles = useStore(state => state.myVehicles);
+	const myFavoriteVehicles = myVehicles.filter(vehicle => vehicle.favorite === true);
+
 	return (
-		<StyledFlex margin="4vh 10px 12vh 10px" gap="40px" alignItems="center">
-			<Card />
-		</StyledFlex>
+		<>
+			<CarFilter checkState={filterState} />
+			<StyledFlex margin="4vh 10px 12vh 10px" gap="40px" alignItems="center">
+				{filterState && <Card data={myFavoriteVehicles} />}
+				{!filterState && <Card data={myVehicles} />}
```

Comment:
An alternative could be: 
```suggestion
				{filterState ? <Card data={myFavoriteVehicles} /> : <Card data={myVehicles} />}
```

Reply:
Yep, implemented since it is shorter :) Thx